sys/include/checksum/crc16_ccitt.h

```diff
@@ -37,8 +34,92 @@
 extern "C" {
 #endif
 
+/**
+ * @brief           Update CRC16-CCITT-KERMIT
+ *
+ * @param[in]  crc  A start value for the CRC calculation, usually the
+ *                  return value of a previous call to
+ *                  crc16_ccitt_kermit_calc(), crc16_ccitt_mcrf4xx_calc() or crc16_ccitt_kermit_update()
+ * @param[in]  buf  Start of the memory area to checksum
+ * @param[in]  len  Number of bytes to checksum
+ *
+ * @return          Checksum of the specified memory area based on the
+ *                  given start value
+ */
+uint16_t crc16_ccitt_kermit_update(uint16_t crc, const unsigned char *buf, size_t len);
+
+/**
+ * @brief           Calculate CRC16-CCITT-KERMIT
+ *                  polynom=0x1021
+ *                  seed=0x0000
+ *                  check=0x2189
+ *
+ * @param[in]  buf  Start of the memory area to checksum
+ * @param[in]  len  Number of bytes to checksum
+ *
+ * @return          Checksum of the specified memory area
+ */
+uint16_t crc16_ccitt_kermit_calc(const unsigned char *buf, size_t len);
+
+/**
+ * @brief           Calculate CRC16-CCITT-MCRF4XX
+ *                  polynom=0x1021
+ *                  seed=0xffff
+ *                  check=0x6f91
+ *
+ * @param[in]  buf  Start of the memory area to checksum
+ * @param[in]  len  Number of bytes to checksum
+ *
+ * @return          Checksum of the specified memory area
+ */
+uint16_t crc16_ccitt_mcrf4xx_calc(const unsigned char *buf, size_t len);
+
+/**
+ * @brief           Update CRC16-CCITT-FALSE
+ *
+ * @param[in]  crc  A start value for the CRC calculation, usually the
+ *                  return value of a previous call to
+ *                  crc16_ccitt_aug_calc(), crc16_ccitt_false_calc() or crc16_ccitt_false_update()
+ * @param[in]  buf  Start of the memory area to checksum
+ * @param[in]  len  Number of bytes to checksum
+ *
+ * @return          Checksum of the specified memory area based on the
+ *                  given start value
+ */
+uint16_t crc16_ccitt_false_update(uint16_t crc, const unsigned char *buf, size_t len);
+
+/**
+ * @brief           Calculate CRC16-CCITT-FALSE
+ *                  polynom=0x1021
+ *                  seed=0xffff
+ *                  check=0x29b1
+ *
+ * @param[in]  buf  Start of the memory area to checksum
+ * @param[in]  len  Number of bytes to checksum
+ *
+ * @return          Checksum of the specified memory area
+ */
+uint16_t crc16_ccitt_false_calc(const unsigned char *buf, size_t len);
+
+/**
+ * @brief           Calculate CRC16-CCITT-AUG
+ *                  polynom=0x1021
+ *                  seed=0x1d0f
+ *                  check=0xe5cc
+ *                  Since CCITT-AUG and CCITT-FALSE only differ in the starting
+ *                  seed, use @ref crc16_ccitt_false_update for updating
```

Comment:
IMHO it makes more sense and easier to understand, if there just would be a wrapper:

```C
/**
 * @brief           Update CRC16-CCITT-AUG
 *
 * @param[in]  crc  A start value for the CRC calculation, usually the
 *                  return value of a previous call to
 *                  crc16_ccitt_aug_calc() or crc16_ccitt_aug_update()
 * @param[in]  buf  Start of the memory area to checksum
 * @param[in]  len  Number of bytes to checksum
 *
 * @return          Checksum of the specified memory area based on the
 *                  given start value
 */
static inline uint16_t crc16_ccitt_aug_update(const unsigned char *buf, size_t len)
{
    /* Since CCITT-AUG and CCITT-FALSE only differ in the starting
     * seed, we wrap around crc16_ccitt_false_update() for updating */
    return crc16_ccitt_false_update(buf, len);
}
```